packages/frontend/components/ActionModal/ActionModal.tsx

```diff
@@ -14,142 +15,41 @@ export default function ActionModal({ toggleActionModal }: ActionModalProps) {
   const handleActionSelection = (action: Action) => {
     addAction(action)
     toggleActionModal(false)
   }
 
   useEffect(() => {
-    // How should we load actions?
-    setActions([
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.TreasureBuidl,
-          operation: Operation.deposit,
-        },
-        input: {
-          token: Token.USDT,
-          quantity: null,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.TreasureBuidl,
-          operation: Operation.withdraw,
-        },
-        output: {
-          token: Token.USDT,
-          quantity: null,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Aave,
-          operation: Operation.deposit,
-        },
-        input: {
-          token: Token.USDT,
-          quantity: null,
-        },
-        output: {
-          token: Token.USDC,
-          quantity: null,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Aave,
-          operation: Operation.withdraw,
-        },
-        input: {
-          token: Token.USDT,
-          quantity: null,
-        },
-        output: {
-          token: Token.USDC,
-          quantity: null,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Aave,
-          operation: Operation.borrow,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Aave,
-          operation: Operation.repay,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Uniswap,
-          operation: Operation.swap,
-        },
-        input: {
-          token: Token.USDC,
-          quantity: null,
-        },
-        output: {
-          token: Token.USDT,
-          quantity: null,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Uniswap,
-          operation: Operation.addLiquidity,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Uniswap,
-          operation: Operation.removeLiquidity,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Compound,
-          operation: Operation.supply,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Compound,
-          operation: Operation.withdraw,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Compound,
-          operation: Operation.repay,
-        },
-      },
-      {
-        id: uuidv4(),
-        type: {
-          protocol: Protocol.Compound,
-          operation: Operation.collectComp,
-        },
-      },
-    ])
+    let actions = [];
+    for (let protocol of libraries) {
+      for (let i = 0; i < protocol.functionSig.length; i++) {
```

Comment:
I think this is good enough but probably we can use a forEach instead of for. Not a blocker but nice to have.